### ISSUE TITLE
Add specification of the overlay schedule to the formal spec

### ISSUE THREAD

**Issue by nc6** [IntersectMBO/cardano-ledger#1524]
Aggelos has requested this, and to request that Christian (and somebody else) reviews it

**Comment by nc6** [comment]
```
-- | Overlay schedule
-- This is just a very simple round-robin, evenly spaced schedule.
overlaySchedule ::
  EpochNo ->
  Set (KeyHash 'Genesis crypto) ->
  PParams ->
  ShelleyBase (Map SlotNo (OBftSlot crypto))
overlaySchedule e gkeys pp = do
  let dval = intervalValue $ _d pp
  if dval == 0
    then pure Map.empty
    else do
      ei <- asks epochInfo
      slotsPerEpoch <- epochInfoSize ei e
      firstSlotNo <- epochInfoFirst ei e
      asc <- asks activeSlotCoeff
      let numActive = dval * fromIntegral slotsPerEpoch
          dInv = 1 / dval
          ascValue = (intervalValue . activeSlotVal) asc
          toRelativeSlotNo x = (Duration . floor) (dInv * fromInteger x)
          toSlotNo x = firstSlotNo +* toRelativeSlotNo x
          genesisSlots = [toSlotNo x | x <- [0 .. (floor numActive - 1)]]
          numInactivePerActive = floor (1 / ascValue) - 1
          activitySchedule = cycle (True : replicate numInactivePerActive False)
          unassignedSched = zip activitySchedule genesisSlots
          genesisCycle = if Set.null gkeys then [] else cycle (Set.toList gkeys)
          active =
            Map.fromList $
              fmap
                (\(gk, (_, s)) -> (s, ActiveSlot gk))
                (zip genesisCycle (filter fst unassignedSched))
          inactive =
            Map.fromList $
              fmap
                (\x -> (snd x, NonActiveSlot))
                (filter (not . fst) unassignedSched)
      pure $ Map.union active inactive
```

**Comment by JaredCorduan** [comment]
In words:

If `d=0` then we are fully decentralized, so we can return the empty schedule.

Otherwise, retrieve the following: the number of slots per epoch, the first slot in the given epoch, and the active slot coefficient.

Next, mark off the `genesisSlots`. These are the slots in the overlay schedule, which will soon be marked as either a PBFT slot or a silent slot. The n-th genesis slot is given by `firstSlotNo + floor( n / d)`, until we run out of slots in the epoch.

Now we designate each genesis slot as either PBFT or silent. The first genesis slot is assigned the first genesis node (so a PBFT slot). Then the next `floor (1/f)-1` genesis slots are marked as silent. The next genesis slot is assigned to the second genesis node, etc.

Depending on the values, it could be that that first genesis node gets one more PBFT slot than the last one, every epoch (but this is proportionally very small).

**Comment by JaredCorduan** [comment]
closed by #1862
